src/utils/ImportPathResolver.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * @file
+ *
+ * Import statements come in a lot of flavors, so having a single
+ * regex that can capture all of those with minimal side effects
+ * is trickly. In this file this regex is constructed from multiple parts.
+ *
+ * Using a named captured group (supported in ES2018/Node 10+)
+ * to allow arbitrary complexity of the regex without worrying
+ * about messing up indexing.
+ *
+ * Meant to match ESM/CommonJS import patterns.
+ *
+ * ⚠ Can match content of strings and comments!
+ *
+ * @example
+ * // Examples of import statements that must be matched
+ * // (Note that there could be newlines between tokens.)
+ * const module = require('some/path')
+ * import module from 'some/path'
+ * import "some/path"
+ * import theDefault, {namedExport} from 'some/path'
+ * const asyncImport = await import('some/path')
+ * export * from 'some/path';
+ */
+
+/** */
+import { existsSync } from 'fs';
+import * as normalizePath from 'normalize-path';
+import { resolve, join, dirname } from 'path';
+
+export type StringReplacer = (importStatement: string) => string;
+
+const anyQuote = '["\'`]';
+const notQuote = '[^"\'`]';
+const importString = `(?:${anyQuote}${notQuote}+${anyQuote})`;
+
+// Separate patterns for each style of import statement,
+// wrapped in non-capturing groups,
+// so that they can be strung together in one big pattern.
+const funcStyle = `(?:\\b(?:import|require)\\s*\\(\\s*${importString}\\s*\\))`;
+const globalStyle = `(?:\\bimport\\s+${importString})`;
+const fromStyle = `(?:\\bfrom\\s+${notQuote}*?${importString})`;
+
+const importRegexString = `(?:${[funcStyle, globalStyle, fromStyle].join(
+  '|'
+)})`;
+
+class ImportPathResolver {
+  constructor(public source: string, readonly sourcePath: string) {}
+
+  get sourceDir() {
+    return dirname(this.sourcePath);
+  }
+
+  /**
+   * Replace all source import paths, using a replacer
+   * function (a la `String.prototype.replace(globalRegex,replacer)`)
+   */
+  replaceSourceImportPaths(replacer: StringReplacer) {
```

Comment:
This is essentially what the code was originally doing to resolve aliases, as a method that allows the dev to not have to also provide the regex.